main.tsp

```diff
@@ -27,5 +27,5 @@ using TypeSpec.Http;
   },
   version: "2.0.0",
 })
-@useAuth(BearerAuth)
+@useAuth(ApiKeyAuth<ApiKeyLocation.query, "AUTHORIZATION">)
```

Comment:
@bterlson, is there a way to tell the ApiKeyAuth piece about a string prefix it needs to add before the API key?

Reply:
https://github.com/microsoft/typespec/issues/2503

Reply:
it is a bearerkey defined in the original api though 

https://github.com/openai/openai-openapi/blob/58fee4955e93bb5762ec612126441330a044f5d2/openapi.yaml#L2199
```
 securitySchemes:
    ApiKeyAuth:
      type: http
      scheme: 'bearer'
```


`ApiKeyAuth` is just a free name they choosed

Reply:
The exact same name can be achieved by using your own model name instead of `BearerAuth` if this is important 

```
import "@typespec/http";

using TypeSpec.Http;

@service
@useAuth(ApiKeyAuth)
namespace DemoService;

model ApiKeyAuth is BearerAuth {}
```
https://cadlplayground.z22.web.core.windows.net/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7Cgp1c2luZyBUeXBlU3BlYy5IdHRwOwoKQHNlcnZpY2UKQHVzZUF1dGgoQXBpS2V5xAspCm5hbWVzcGFjZSBEZW1vU8YrOwoKbW9kZWwgyiogaXMgQmVhcmVyxQ57fQ%3D%3D&e=@typespec/openapi3&options=%7B%22linterRuleSet%22:%7B%22extends%22:%5B%22@typespec/http/all%22%5D%7D%7D